### PR TITLE
[EMB-316] Cookie consent banner

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
Notify logged-out users that the site uses cookies, and prompt them to consent.

## Summary of Changes
- Add `cookie-banner` component
  - Displays for logged-out users without an `osf_cookieconsent` cookie
  - Added to home page, institutions landing page
- Enable `join-osf-banner` on node pages
  - Fix it up a little so it works with e-css-modules

## Side Effects / Testing Notes
The new banner overlaps the "join osf" banner. This is consistent with the other cookie banner implementations, but bears noting:
bannerbanner

## Ticket
https://openscience.atlassian.net/browse/EMB-316

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: config/environment.d.ts [hunk 1]
```diff
@@ -78,6 +78,9 @@ declare const config: {
         authenticator: string;
         keenProjectId?: string;
         analyticsDismissAdblockCookie: string;
+        cookies: {
+            cookieConsent: string,
+        },
```
Comment (brianjgeiger) [inline config/environment.d.ts]: Drive-by: There are some more cookies on L76-77 that should be wrapped up into this if you're going to have it. The names should be normalized as well.

Comment (aaxelb) [reply config/environment.d.ts]: Yeah, I got those in [another PR](https://github.com/CenterForOpenScience/ember-osf-web/pull/331/files#diff-695693c5dabaffb064bde5b060195ff7R154), so I figured I'd follow that pattern here and make the conflicts easy to deal with.

The others don't start with `cookie`, but since this one is a metacookie, a cookie indicating consent to cookies, it seemed clearer to name it that way.

File: lib/osf-components/addon/components/cookie-banner/template.hbs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{{#if showBanner}}
+    <div local-class="Buffer"></div>
```
Comment (jamescdavis) [inline lib/osf-components/addon/components/cookie-banner/template.hbs]: Does this do anything?

Comment (aaxelb) [reply lib/osf-components/addon/components/cookie-banner/template.hbs]: Oh whoops, nope.

File: lib/osf-components/addon/components/cookie-banner/template.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    <div local-class="CookieBanner">
+        <div local-class="Warning">
+            {{t 'cookieBanner.cookieWarning'}}
+        </div>
+        <div local-class="Accept">
+            <button class="btn btn-default" {{action 'accept'}}>
```
Comment (jamescdavis) [inline lib/osf-components/addon/components/cookie-banner/template.hbs]: `bs-button`?
